src/main/scala/it/pagopa/interop/agreementprocess/api/impl/AgreementApiServiceImpl.scala

```diff
@@ -736,6 +744,29 @@ final case class AgreementApiServiceImpl(
     } yield updated
   }
 
+  private def extractArchivables(
+    activatingAgreementId: UUID,
+    agreements: Seq[AgreementManagement.Agreement]
+  ): Seq[AgreementManagement.Agreement] =
+    agreements.filter(a => ARCHIVABLE_STATES.contains(a.state) && a.id == activatingAgreementId)
```

Comment:
doesn't this extract just the agreement corresponding to `activatingAgreementId` ?

Reply:
You're right.
Thanks!